src/main/java/de/tum/in/www1/artemis/config/migration/entries/BambooMigrationService.java

```diff
@@ -0,0 +1,499 @@
+
+        var response = restTemplate.exchange(builder.build().toUri(), HttpMethod.GET, null, String.class);
+        var html = response.getBody();
+        var ids = getDataItemIds(html);
+        for (var id : ids) {
+            var repositoryName = getRepositoryNameById(html, id);
+            // old repositories have the name "Assignment", new ones "assignment"
+            if (repositoryName.isPresent() && repositoryName.get().equalsIgnoreCase(name)) {
+                return Optional.of(id);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void overrideRepositoriesToCheckout(String buildPlanKey, List<AuxiliaryRepository> auxiliaryRepositoryList) {
+        Optional<Long> testRepositoryId = getConnectedRepositoryId(buildPlanKey, "tests");
+        Optional<Long> assignmentRepositoryId = getConnectedRepositoryId(buildPlanKey, "assignment");
+
+        if (testRepositoryId.isEmpty()) {
+            log.error("Repository tests not found for build plan {}", buildPlanKey);
+        }
+        if (assignmentRepositoryId.isEmpty()) {
+            log.error("Repository assignment not found for build plan {}", buildPlanKey);
+        }
+
+        Map<String, Long> auxiliaryRepositoryIds = new HashMap<>();
+        for (AuxiliaryRepository auxiliaryRepository : auxiliaryRepositoryList) {
+            Optional<Long> repositoryId = getConnectedRepositoryId(buildPlanKey, auxiliaryRepository.getName());
+            if (repositoryId.isPresent()) {
+                auxiliaryRepositoryIds.put(auxiliaryRepository.getName(), repositoryId.get());
+            }
+            else {
+                log.error("Repository {} not found for build plan {}", auxiliaryRepository.getName(), buildPlanKey);
+            }
+        }
+        setRepositoriesToCheckout(buildPlanKey, testRepositoryId.orElseThrow(), assignmentRepositoryId.orElseThrow(), auxiliaryRepositoryIds, auxiliaryRepositoryList);
+    }
+
+    @Override
+    public Page<ProgrammingExerciseStudentParticipation> getPageableStudentParticipations(
+            ProgrammingExerciseStudentParticipationRepository programmingExerciseStudentParticipationRepository, Pageable pageable) {
+        return programmingExerciseStudentParticipationRepository.findAllWithBuildPlanId(pageable);
+    }
+
+    @Override
+    public boolean supportsAuxiliaryRepositories() {
+        return true;
+    }
+
+    @Override
+    public boolean buildPlanExists(String projectKey, String buildPlanKey) {
+        if (projectKey == null || buildPlanKey == null) {
+            return false;
+        }
+        return bambooService.checkIfBuildPlanExists(projectKey, buildPlanKey);
+    }
+
+    /**
+     * Returns a list of all notification ids for the given build plan for this server.
+     * Bamboo doesn't provide a REST endpoint for this, so we have to parse the HTML page using the admin chain API.
```

Comment:
I can only test the form for the currently supported Confluence version of Artemis, if the form changes then I can't promise that it will work with future releases.